eofConvention.fsx

```diff
@@ -0,0 +1,30 @@
+open System.IO
+
+let EolAtEof(path: FileInfo) = 
+
+    use streamReader = new StreamReader (path.FullName)
+    let filetext = streamReader.ReadToEnd()
+    
+    if filetext <> "" then
+        (filetext |> Seq.last = '\n')
+    else
+        true
+
+let NotPrefix (prefix: string) (pathStr: string) =
+    not (pathStr.StartsWith(prefix))
+
+let invalidFiles = 
+    Directory.GetFiles(".", "*.*", SearchOption.AllDirectories) 
+    |> Seq.filter (NotPrefix "./node_modules")
+    |> Seq.filter (NotPrefix "./.git")
+    |> Seq.map (fun pathStr -> FileInfo pathStr)
```

Comment:
@realmarv if you move Seq.map to be the first call after Directory.GetFiles then you do less Primitive Obsession